draft-dekater-scion-dataplane.md

```diff
@@ -1342,20 +1355,97 @@ This section describes the steps that a SCION egress border router must perform
+Also in SCION, an on-path adversary can always simply drop packets. This kind of attack is fundamental and generally cannot be prevented. However, in this case, the endpoint can use SCION's path-awareness to immediately select an alternate path if available.
+
+**Note:**<br>
+For more information on the SCION path header, see [](#path-header).<br>
+For more information on the SCION address header, see [](#address-header).
+
+
+#### Modification of the Path Header
+
+It is possible for an on-path adversary to modify the SCION path header of a packet, and replace the remaining part of path segments  to the destination with different segments. Such replaced segments must include authorized segments, as otherwise the packet would be simply dropped at some point. The already traversed portion of the current segment and past segments can be modified by the adversary in a completely arbitrary manner (for instance, deleting and adding valid and invalid hop fields). On reply packets from the destination, the adversary can transparently revert the changes to the path header again. For instance, if an adversary M is an intermediate AS on the path of a packet from A to B, then M can replace the packet’s past path (leading up to, but not including M). The new path may not be a valid end-to-end path. However, when B reverses the path and sends a reply packet, that packet would go via M, which can then transparently change the invalid path back to the valid path to A.
+
+Currently, SCION as described in this document does not provide this type of protection. Endpoints may optionally uses the SCION Packet Authentication Option (SPAO) extension that provides integrity protection for the path header, allowing detection of such attack. This extension is described in [CHUAT22].
+
+Moreover, packet integrity protection is not enough if there are two colluding adversaries on the path. These colluding adversaries can forward the packet between them using a different path than selected by the source endpoint: The first on-path attacker remodels the packet header arbitrarily, and the second on-path attacker changes the path back to the original source-selected path, such that the integrity check by the destination endpoint succeeds. To prevent this attack and to defend against multiple on-path adversaries in general, path validation is required (which is currently not supported by SCION).
+
+
+#### Modification of the Address Header
+
+Besides modifying the SCION path header, an on-path adversary can also modify the source and destination addresses in the SCION address header. If an adversary changes the address of the destination endpoint, it needs to modify the "future" path between itself and the new destination accordingly; otherwise the packet would simply be dropped by the last AS on the path.
+
+Modifications of the SCION address header can be discovered by the destination endpoint if the packet is integrity protected (which is currently not supported by SCION).
```

Comment:
```suggestion
Modifications of the SCION address header can be discovered by the destination endpoint if the packet is integrity protected with the SPAO extension. Currently, SCION as described in this document does not provide this type of protection.
```

SPAO also MACs the endpoint address, right? (Yes, according to [docs](https://docs.scion.org/en/latest/protocols/authenticator-option.html#authenticated-data)).

Reply:
True. I think the SPAO was left out intentionally. Makes sense to allude to its existence, but maybe it would be good to be a bit more generic.

> Modifications of the SCION address header can be discovered by the destination endpoint by a data integrity protection system. Such a data integrity protection system, losely analogous to the IPSec Authentication Header, exists for SCION but is out of scope for this document.